pkg/plugin/utils.go

```diff
@@ -57,24 +57,24 @@ func collapseToSingleLine(query string) string {
 	return regexp.MustCompile("[\r\n\t]+").ReplaceAllString(query, " ")
 }
 
-// Convert the value of the "_time" field to an ISO timestamp string
-func timeToIsoString(timeValue interface{}) (bool, string) {
+// Convert the value of the "_time" field (expected to be in seconds) to a time.Time in UTC
+func criblTimeToGrafanaTime(timeValue interface{}) (bool, time.Time) {
 	var seconds float64
 	switch v := timeValue.(type) {
 	case float64:
 		seconds = v
 	case string:
 		s, err := strconv.ParseFloat(v, 64)
 		if err != nil {
-			return false, ""
+			return false, time.Time{}
 		}
 		seconds = s
 	default:
-		return false, ""
+		return false, time.Time{}
 	}
 	wholeSec := int64(seconds)
-	nanoSec := int64(math.Round((seconds-float64(wholeSec))*1000.0)) * 1000000 // ms precision
-	return true, time.Unix(wholeSec, nanoSec).UTC().Format(time.RFC3339Nano)
+	nanoSec := int64(math.Round((seconds-float64(wholeSec))*1000000.0)) * 1000 // microsec precision
+	return true, time.Unix(wholeSec, nanoSec).UTC()
```

Comment:
NOTE: I changed this to microsec because we can, and it provides additional granularly should it be needed.  I tried nanoseconds and was getting some floating point precision garbage (i.e. 122999 instead of 123000), so I figured this is good enough for ~100% of the likely scenarios.